External/HIP/utils/log_data.py

```diff
@@ -0,0 +1,72 @@
+#!/bin/python3
+import argparse
+import os
+import csv
+import time
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Record data and calculate statistics.')
+    parser.add_argument('--data', type=float, required=True, help='The data value to record.')
+    parser.add_argument('--log-file', type=str, required=True, help='The file to log data to.')
+    parser.add_argument('--label', type=str, required=True, help='The label for the data.')
+    parser.add_argument('--time-stamp', type=str, required=False, help='The timestamp for the data.')
+    
+    args = parser.parse_args()
+    return args
+
+def read_existing_data(file_name):
+    data = []
+    if os.path.exists(file_name) and os.path.getsize(file_name) > 0:
+        with open(file_name, 'r') as file:
+            reader = csv.reader(file)
+            for row in reader:
+                if row and row[2].strip():
+                    try:
+                        data.append(float(row[2].strip()))
+                    except ValueError:
+                        continue
+    return data
+
+def calculate_average(data):
```

Comment:
The tests already depend on numpy. You could just use numpy to do the math,

Reply:
The script may be used alone. Since the calculation is trivial, it is unnecessary to introduce extra dependency.